### PR TITLE
whip/whep example

### LINKED ISSUE #2499: Example: using WHIP and WHEP
## Summary

Example: using WHIP and WHEP instead of SDP pasted

## Motivation

SDP paste sucks. Using whip (OBS) / whep make example more easy for learning.

### PR DESCRIPTION
**Pull request by stephanrotolante**
#### Description
Small example of a WHIP/WHEP exchange

#### Reference issue
resolves #2499

### CODE REVIEW

**Comment by stephanrotolante** [comment]
@Sean-Der Let me know if you'd like to see more implemented. 

There a a few more status codes and http verbs as part of a full WHIP/WHEP implementation

File: examples/whip-whep/main.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	"github.com/pion/webrtc/v4"
+)
+
+var (
+	videoTrack  *webrtc.TrackLocalStaticRTP
+	rtpListener *net.UDPConn
+	err         error
+)
+
+const MTU_SIZE = 1600
```
Comment (daonb) [inline examples/whip-whep/main.go:27]: Please add a comment as the linter  requests

File: examples/whip-whep/main.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	"net"
+	"net/http"
+
+	"github.com/pion/webrtc/v4"
+)
+
+var (
+	videoTrack  *webrtc.TrackLocalStaticRTP
+	rtpListener *net.UDPConn
+	err         error
```
Comment (daonb) [inline examples/whip-whep/main.go:24]: Global error is not a good idea and it also raises the linter's wrath on line 30

**Review by daonb — CHANGES_REQUESTED** [review]
Thanks Stephan! The code looks clean and simple.

**Comment by Sean-Der** [comment]
Great work @stephanrotolante ! We should be able to merge this super quick

Instead of FFMPEG can we do a flow with OBS? The newest version has WHIP support and is what people want to build with.

I think the goal is publish once with OBS then watch unlimited times.

File: examples/whip-whep/main.go (hunk 1 above)
Comment (stephanrotolante) [reply examples/whip-whep/main.go:27]: good call out will fix this

File: examples/whip-whep/main.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
+// SPDX-FileCopyrightText: 2023 The Pion community <https://pion.ly>
+// SPDX-License-Identifier: MIT
+
+//go:build !js
+// +build !js
+
+// whip-whep demonstrates the WHIP/WHEP specifications to exhange SDP descritpions
```
[the remark this answers is not in the document]
Comment (stephanrotolante) [reply examples/whip-whep/main.go:7]: Thanks for the feedback!

I will update those comments. 

You are correct this does fully implement whip/whep, but I will leave that out for now as this is just a crude example to integrate into the new obs whip/whep source

**Comment by Sean-Der** [comment]
@stephanrotolante I made some changes, mind checking them out?

I put `WHIP` and `WHEP` in the same example. I was able to publish via OBS, and then pull back the same feed via WHEP.

```
http://localhost:8080/whip
```

and 
```
http://localhost:8080/whep
```

Since your name is on the commit mind trying out and make sure you like it/agree with how it all flows? I am going to update the docs tomorrow and merge!

**Comment by stephanrotolante** [comment]
> @stephanrotolante I made some changes, mind checking them out?
> 
> I put `WHIP` and `WHEP` in the same example. I was able to publish via OBS, and then pull back the same feed via WHEP.
> 
> ```
> http://localhost:8080/whip
> ```
> 
> and
> 
> ```
> http://localhost:8080/whep
> ```
> 
> Since your name is on the commit mind trying out and make sure you like it/agree with how it all flows? I am going to update the docs tomorrow and merge!

Looks good to me!

**Comment by stephanrotolante** [comment]
@Sean-Der with my readme I added a video (there is no audio), walking through how to use OBS with a WHEP source. You might want to delete that since the api was changed up. It looks like I got WHEP + WHIP mixed up

**Comment by Sean-Der** [comment]
@stephanrotolante Done! Merged into master, great work :)

Give it a try and I am sure I made mistakes, please send PRs and will merge them quick!

**Pull request merged**